### PR TITLE
GLT-1241 update LibraryDesign

### PR DESCRIPTION
**Pull request by callunity**
- add LibraryDesignCode to LibraryAdditionalInfo and LibraryDesign

### CODE REVIEW

File: migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/destination/ValueTypeLookup.java [hunk 1]
```diff
@@ -751,6 +781,10 @@ public void resolveAll(Library library) throws IOException {
         if (ld == null) throw new IOException("LibraryDesign not found");
         lai.setLibraryDesign(ld);
       }
+
+      LibraryDesignCode ldc = resolve(lai.getLibraryDesignCode());
+      if (ldc == null) throw new IOException("LibraryDesignCode not found");
```
Comment (djcooke) [inline migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/destination/ValueTypeLookup.java]: Please clarify message (null not allowed)

File: sqlstore/src/main/resources/db/migration/V8203__libraryDesignCode_changes.sql [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+-- add current LibraryDesignCode values into the table
+INSERT INTO LibraryDesignCode (code, description) SELECT DISTINCT SUBSTRING(name, 1,2), SUBSTRING(name, 1,2) FROM LibraryDesign;
+
+ALTER TABLE LibraryDesign DROP FOREIGN KEY `FK_ld_lt`;
+ALTER TABLE LibraryDesign DROP COLUMN libraryType;
+ALTER TABLE LibraryAdditionalInfo ADD COLUMN `libraryDesignCodeId` BIGINT(20);
+ALTER TABLE LibraryAdditionalInfo ADD CONSTRAINT `FK_lai_libraryDesignCode_libraryDesignCodeId` FOREIGN KEY (`libraryDesignCodeId`) REFERENCES `LibraryDesignCode` (`libraryDesignCodeId`);
+ALTER TABLE LibraryDesign ADD COLUMN `libraryDesignCodeId` BIGINT(20);
+ALTER TABLE LibraryDesign ADD CONSTRAINT `FK_ld_libraryDesignCode_libraryDesignCodeId` FOREIGN KEY (`libraryDesignCodeId`) REFERENCES `LibraryDesignCode` (`libraryDesignCodeId`);
+--ALTER TABLE LibraryDesign DROP CONSTRAINT `uk_libraryDesign_name`;
```
Comment (djcooke) [inline sqlstore/src/main/resources/db/migration/V8203__libraryDesignCode_changes.sql]: Is this supposed to be commented out? If so, remove instead

Comment (djcooke) [inline sqlstore/src/main/resources/db/migration/V8203__libraryDesignCode_changes.sql]: There was supposed to be a unique key on name+sampleclass too, right?

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/LimsBindingInitializer.java [hunk 3]
```diff
@@ -678,11 +683,29 @@ public void setAsText(String element) throws IllegalArgumentException {
         if (id == -1) {
           setValue(null);
         } else {
-          setValue(libraryDesignDao.getLibraryDesign(id));
+          try {
+            setValue(libraryDesignDao.getLibraryDesign(id));
+          } catch (IOException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
```
Comment (apmasell) [inline spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/LimsBindingInitializer.java]: This should go through log4j and the value set to `null`.

File: migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/destination/ValueTypeLookup.java (hunk 1 above)
Comment (apmasell) [inline migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/destination/ValueTypeLookup.java]: And include the original design code that failed.

Comment (djcooke) [inline migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/destination/ValueTypeLookup.java]: Oh, I misread this. Thought it was a case of `lai.getLibraryDesignCode()` being null. Disregard my comment

**Pull request merged**